contracts/exchange/contracts/src/MixinMatchOrders.sol

```diff
@@ -263,6 +259,473 @@ contract MixinMatchOrders is
+                matchResults.right.takerAssetFilledAmount
+            );
+
+            // Aggregate the new fill results with the previous fill results for the current orders.
+            _addFillResults(
+                leftFillResults,
+                matchResults.left
+            );
+            _addFillResults(
+                rightFillResults,
+                matchResults.right
+            );
+
+            // Update the profit in the left and right maker assets using the profits from
+            // the match.
+            batchMatchedFillResults.profitInLeftMakerAsset = _safeAdd(
+                batchMatchedFillResults.profitInLeftMakerAsset,
+                matchResults.profitInLeftMakerAsset
+            );
+            batchMatchedFillResults.profitInRightMakerAsset = _safeAdd(
+                batchMatchedFillResults.profitInRightMakerAsset,
+                matchResults.profitInRightMakerAsset
+            );
+
+            // If the leftOrder is filled, update the leftIdx, leftOrder, and leftSignature,
+            // or break out of the loop if there are no more leftOrders to match.
+            if (leftOrderInfo.orderTakerAssetFilledAmount >= leftOrder.takerAssetAmount) {
+                // Update the batched fill results once the leftIdx is updated.
+                batchMatchedFillResults.left[leftIdx++] = leftFillResults;
+                // Clear the intermediate fill results value.
+                leftFillResults = LibFillResults.FillResults(0, 0, 0, 0);
+
+                // If all of the left orders have been filled, break out of the loop.
+                // Otherwise, update the current right order.
+                if (leftIdx == leftOrders.length) {
+                    // Update the right batched fill results
+                    batchMatchedFillResults.right[rightIdx] = rightFillResults;
+                    break;
+                } else {
+                    leftOrder = leftOrders[leftIdx];
+                    leftOrderInfo = getOrderInfo(leftOrder);
+                }
+            }
+
+            // If the rightOrder is filled, update the rightIdx, rightOrder, and rightSignature,
+            // or break out of the loop if there are no more rightOrders to match.
+            if (rightOrderInfo.orderTakerAssetFilledAmount >= rightOrder.takerAssetAmount) {
+                // Update the batched fill results once the rightIdx is updated.
+                batchMatchedFillResults.right[rightIdx++] = rightFillResults;
+                // Clear the intermediate fill results value.
+                rightFillResults = LibFillResults.FillResults(0, 0, 0, 0);
+
+                // If all of the right orders have been filled, break out of the loop.
+                // Otherwise, update the current right order.
+                if (rightIdx == rightOrders.length) {
+                    // Update the left batched fill results
+                    batchMatchedFillResults.left[leftIdx] = leftFillResults;
```

Comment:
Is this necessary if the left order was also filled during this iteration? I believe this would write an empty fill result for the next iteration's `left` order. If this is the case, then we should ensure there is a test case for this scenario.

Reply:
Good point. I actually don’t know if this is a problem though since in this case the next left order wouldn’t have been filled anyways. It’s redundant, but it doesn’t seem to change any behavior since we process the orders in order and the last order will never be filled in this case. I actually think it may be cleaner to keep it the same, but I’m interested to hear your thoughts.

Reply:
Please see comment above 🙏